compiler/resolution/preFold.cpp

```diff
@@ -1123,14 +1123,27 @@ static Expr* preFoldNamed(CallExpr* call) {
         USR_FATAL(call, "illegal call of type");
       }
 
+      if (call->numActuals() > 3) {
+        USR_FATAL(call, "too many arguments to type index expression");
+      }
+
       if (!get_int(call->get(3), &index)) {
         USR_FATAL(call, "illegal type index expression");
       }
 
-      if (!isAggregateType(sym->type)) {
+      AggregateType* at = toAggregateType(sym->type);
+
+      if (!at) {
         USR_FATAL(call, "illegal type index expression");
       }
 
+      if (index <= 0 || index > at->fields.length-1) {
+        char msg[60];
+        snprintf(msg, 60,
+                 "type index expression '%" PRId64 "' out of bounds", index);
+        USR_FATAL(call, msg);
```

Comment:
Shouldn't this just be USR_FATAL(call, "type index expression '%i out of bounds", (int) index) ?

Reply:
Yes, that sounds better to me.  I'll update it.